### PR TITLE
Sheriff profile

### PR DESCRIPTION
**Pull request by hollyegrayson**
Added functionality to create a sheriff profile and view a sheriff profile

Disabling the timeline for time being, as per @cjam request

### CODE REVIEW

File: src/app/components/Form/FormFieldWrapper.tsx [hunk 1]
```diff
@@ -4,16 +4,17 @@ import { WrappedFieldProps } from 'redux-form';
...
 export default class FormFieldWrapper extends React.PureComponent<FormFieldWrapperProps>{
     render(){
-        const {meta:{touched, error, warning}, label} = this.props;
+        const {meta:{touched, error, warning}, label, controlId} = this.props;
         const {children} = this.props;
         const validationState = touched ? (error ? 'error': (warning ? 'warning': undefined)) : undefined;
         
         return (
-            <FormGroup validationState={validationState}>
+            <FormGroup validationState={validationState} controlId={controlId}>
```
Comment (cjam) [inline src/app/components/Form/FormFieldWrapper.tsx]: I don't see the controlId being used anywhere but if we need one could we perhaps use something like 
```ts
`${formname}-${fieldName}
```

File: src/app/modules/sheriffs/components/SheriffGrid.tsx [hunk 2]
```diff
@@ -22,9 +25,25 @@ class SheriffGrid extends React.PureComponent<SherriffGridProps, any>{
                         ))}
                     </Row>
                 </Grid>
+                {/* <ListGroup>
```
Comment (cjam) [inline src/app/modules/sheriffs/components/SheriffGrid.tsx]: can remove

File: src/app/modules/sheriffs/components/SheriffProfileView.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+export interface SheriffProfileViewProps {
+    sheriff: Sheriff;
+}
+
+export default class SheriffProfileView extends React.Component<SheriffProfileViewProps, any>{
+    render() {
+        const { sheriff } = this.props;
+        return (
+            <div>
```
Comment (cjam) [inline src/app/modules/sheriffs/components/SheriffProfileView.tsx]: might improve readability a bit without the whitespace, ironically enough:

```tsx
<div>
       <WorksiteDetailsView sheriff={ sheriff } />
        <br />
        <TrainingDetailsView sheriff={ sheriff } />
        <br />
        <UpcomingScheduleView />
        <br /> 
        <OvertimeDetails />
</div>
```

File: package.json [hunk 4]
```diff
@@ -15,18 +15,21 @@
...
-    "react-bootstrap": "^0.31.5",
     "react-calendar-timeline": "^0.15.2",
+    "react-bootstrap": "^0.32.0",
+    "react-datetime": "^2.11.1",
     "react-dev-utils": "^4.0.1",
     "react-dnd": "^2.5.4",
     "react-dnd-html5-backend": "^2.5.4",
     "react-dom": "^16.1.1",
+    "react-image-upload": "0.0.1",
+    "react-images-upload": "^1.1.1",
```
[the remark this answers is not in the document]
Comment (hollyegrayson) [reply package.json]: done

File: src/app/api/index.ts [hunk 5]
```diff
@@ -12,11 +12,39 @@ export enum SheriffAbility {
...
-    badgeNumber: number
-    imageUrl?: string
-    abilities: SheriffAbility
+    firstName: string;
+    lastName: string;
+    badgeNumber: number;
+    imageUrl?: string;
+    abilities: SheriffAbility;
+    training: [{
+        trainingType: string;
```
[the remark this answers is not in the document]
Comment (hollyegrayson) [reply src/app/api/index.ts]: Good idea - done

File: src/app/components/Form/FormFieldWrapper.tsx (hunk 1 above)
Comment (hollyegrayson) [reply src/app/components/Form/FormFieldWrapper.tsx]: We don't need it at this point; I forgot I had added that in - it has been removed

File: src/app/modules/sheriffs/components/SheriffGrid.tsx (hunk 2 above)
Comment (hollyegrayson) [reply src/app/modules/sheriffs/components/SheriffGrid.tsx]: done

File: src/app/modules/sheriffs/components/SheriffProfileView.tsx (hunk 3 above)
Comment (hollyegrayson) [reply src/app/modules/sheriffs/components/SheriffProfileView.tsx]: fixed

File: src/app/components/Navigation.tsx [hunk 6]
```diff
@@ -25,6 +25,7 @@ export default class Navigation extends React.Component<NavigationProps, any>{
                         <NavigationLink exactMatch to="/" label="Home" />
                         <NavigationLink to="/settings" label="Settings" />
                         <NavigationLink to="/timeline" label="Timeline" />
+                        <NavigationLink to="managesheriffs" label="Manage Sheriffs" />
```
[the remark this answers is not in the document]
Comment (hollyegrayson) [reply src/app/components/Navigation.tsx]: done

**Pull request merged**